pkg/upgrade/plan/plan.go

```diff
@@ -94,6 +97,24 @@ func RegisterHandlers(ctx context.Context, serviceAccountName, controllerNamespa
 		return obj, nil
 	})
 
+	// secret events referred to by a plan (potentially) trigger that plan
+	secrets.OnChange(ctx, controllerName, func(key string, obj *corev1.Secret) (*corev1.Secret, error) {
+		if planList, err := plans.Cache().List(controllerNamespace, labels.Everything()); err != nil {
+			logrus.Error(err)
```

Comment:
shouldn't this return nil, err

Reply:
That is a good question. I did intentionally write this secrets-trigger-plan logic to make a best effort attempt and I figured if listing secrets is causing problems it is probably a permissions problem and therefore not something likely to resolve itself (thus the re-enque implied by returning an error seemed gratuitous). I am open to changing this.

Reply:
I am just wondering what will be the result of returning nil object for both the secret and the error to the controller when a listing error happen?

Reply:
It should have fell through to the last line of the function that was returning `obj, nil`. I ran your comment by @ibuildthecloud and he agreed so I changed both the secret and node OnChange events to return the potential error when listing plans to (potentially) trigger